### PR TITLE
Rework front matter

### PR DESCRIPTION
**Pull request by itowlson**
The substance of this PR is:

* Slim down the Introduction page to a minimal "what Spin is" and key links.  (Previously it dived heavily into Rust code and the manifest, which kind of confused the issue with other pages, seemed a bit low-level, and buried the links.)  This may have gone too far - I would like to hear what folks think.
* Rework the quickstart to show specific instructions in JS, Rust, Python and Go, rather than All Rust All The Time.  ~This is incomplete; I want agreement on it before I slog through the remaining languages.~
* Remove dynamic config from the `Configuration` document and move it to a new document in the Advanced section (for now).
* Rename what's left of the Configuration document to "Writing Spin Applications."  (This is too similar to "Developing Spin Applications."  But I intend this to be a transient situation - I just didn't want to rewrite EVERYTHING in one PR.)

Please run it up and kick the tyres!

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [x] Has run `npm run build-index`
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors

### CODE REVIEW

File: content/spin/quickstart.md
Comment (karthik2804) [inline content/spin/quickstart.md:74]: Could we just make this a regular paragraph and not a list.
```suggestion
[See more options for installing Spin](install)
```

**Comment by karthik2804** [comment]
Also open to discussion on the way we present the configuration options, I feel like we should possibly present the information in the form of a table or possibly break it down a bit more if possibly.

File: content/spin/quickstart.md
Comment (itowlson) [reply content/spin/quickstart.md:74]: That's probably more consistent.  I kind of went back and forth on making things stand out vs making them fit in.  Thanks.

**Comment by karthik2804** [comment]
I personally like the direction that this PR is guiding the docs towards. I like the quickstart and on that note, I would like to bring up the ordering for the different languages.  Do we want to stick to alphabetical order or something along the lines of most used language or what I am looking for is some guideline on how we can be consistent across the docs?

File: content/spin/dynamic-configuration.md
Comment (tpmccallum) [inline content/spin/dynamic-configuration.md:3]: I don't see the `enable_shortcodes = true` here.
Perhaps I am confused because I thought `enable_shortcodes = true` was necessary for the `<!-- @nocpy -->` and `<!-- @selectiveCpy -->` to work. But this functionality seems to be working on this page during my local preview without the frontmatter part.

Screenshot 2023-03-02 at 11 01 27 am

@karthik2804 what am I missing here? :)

**Comment by itowlson** [comment]
@karthik2804 when you say the configuration options, are you referring to the manifest?  (sorry I am not trying to be pedantic here I genuinely do not know what 'configuration' refers to in any given case)  If so, yes, absolutely, but I am intentionally treating that as out of scope for this.  I don't want to rework too many pages at once, or it will be hard to review and agree them in a granular way, so for this PR I wanted to do the minimal change to make it make sense as a core dev link - hope that makes sense.  Next PR though...

image

**Comment by itowlson** [comment]
@karthik2804 I was thinking switching to alphabetical order would be the most sustainable (and politically neutral) solution.  Though again I was treating that as out of scope for this PR.

**Comment by tpmccallum** [comment]
This is looking great @itowlson 
No linting (potential CI) errors or `bart check` issues, and the local preview looks great also.

Screenshot 2023-03-02 at 11 11 03 am
Screenshot 2023-03-02 at 11 10 34 am

File: content/spin/dynamic-configuration.md
Comment (karthik2804) [reply content/spin/dynamic-configuration.md:3]: You don't need the enable shortcodes for the code block annotations. Currently, it is only required for `multi-block` content and `details` shortcodes.

**Comment by karthik2804** [comment]
@itowlson, yes I mean the manifest section. It makes sense to work on it in a separate PR and alphabetical order for the code blocks sounds good to me.

File: content/spin/configuration.md
Comment (itowlson) [reply content/spin/configuration.md:36]: YOU MEAN I DON'T HAVE TO MAINTAIN IT MANUALLY?!?!?!?!?!

Man.  I should _totally_ read that contributing guide.  Thanks for the tip.

File: content/spin/dynamic-configuration.md
Comment (tpmccallum) [reply content/spin/dynamic-configuration.md:3]: Ah, for multi-language codeblock; got it!
Thanks

**Comment by itowlson** [comment]
@tpmccallum @karthik2804 I don't know why the link checker is looking in GH main for this:

```
BROKEN─ https://github.com/fermyon/developer/blob/main/content/spin/dynamic-configuration.md
```

The page exists in the PR and I _believe_ the links to it are correct...

**Comment by karthik2804** [comment]
The link will be broken until it is merged into main as that particular link will not exist at this point in time @itowlson. This is expected behavior for new pages.

**Comment by itowlson** [comment]
So how do we get it to do the Check Docs part of the checks?

**Comment by karthik2804** [comment]
Good point. Currently, we have been working on new pages with manual testing. Might be a good idea to move check docs above the npm test.

**Review by tpmccallum — APPROVED** [review]
LGTM, had a good look through all areas and could not see any standout issues.

I like:
- the clarity i.e. that Rust Log is for applications written in all lanauges.

extra notes:
- the only tiny thing that made me think a little is that there is a mention of the `--runtime-config-file` (in the Vault Config Provider Example, of the `dynamic-configuration.md` file) and I see that in the current CLI documents this only as a "trigger" option. Just an observation that there is no other mention of "trigger" on that `dynamic-configuration.md`. This might be fine.

**Comment by tpmccallum** [comment]
The CI error is just because of the above previous discussion.
We have since merged the update https://github.com/fermyon/developer/pull/409 that runs the docs test before the npm test.
I don't think this problem will surface any issues if this is merged. The document failing the broken link check will appear once this is merged.

**Comment by mikkelhegn** [comment]
Is it a problematic (by any dimension) to have the URL `/configuration` now be a document with the title `Writing Spin Applications?` - SEO, existing links, etc? I think we can add a forwarder from old url to new if we decide to change it?

**Comment by mikkelhegn** [comment]
Agree that the manifest reference in the `/configuration` article should be improved. A table format is ususally the best solution is such cases. #413

**Comment by mikkelhegn** [comment]
Any thoughts on what to do with this article? `/developing` - It seems to fall in-between chairs with this change? I noticed a note in the beginning, but not sure what the plan is for the content.

**Comment by mikkelhegn** [comment]
What's the considerations for not including C#/Dotnet together with the other languages?

**Review by mikkelhegn — APPROVED** [review]
This is really good @itowlson! I can approve whatever you do with it from hereon. :-)

**Comment by itowlson** [comment]
@tpmccallum re

> the current CLI documents this only as a "trigger" option. Just an observation that there is no other mention of "trigger" on that dynamic-configuration.md.

From a user point of view it's just a flag they pass to `up`.  The trigger thing is an internal implementation detail; I think bringing up triggers on the dynamic config page just to talk about this would be a distraction.  But thanks for double-checking on it!

**Comment by itowlson** [comment]
@mikkelhegn 

> Is it a problematic (by any dimension) to have the URL /configuration now be a document with the title Writing Spin Applications? - SEO, existing links, etc?
>
> Any thoughts on what to do with this article? /developing - It seems to fall in-between chairs with this change?

Agree.  My next thing is to rework those two pages with new titles, URLs, structures, and breakdowns of responsibility.  But I didn't want to tackle all of it in one monster PR; this PR focuses on the front matter, and makes the least changes it can to the rest of the site to permit delegation out from that front matter.  So hopefully this situation is transient, and _hopefully_ not prohibitively bad compared to where it is now.  Hopefully.

File: content/spin/quickstart.md
Comment (itowlson) [reply content/spin/quickstart.md:93]: I can avoid _showing_ the other-language templates in this screencap, but for now the Rust and Go templates are in together and that's unlikely to change in the short term.  (I know it is on our minds for the medium term.)

Re the tab headers, I wanted to have _specific per-language tabs_ with the name on them, even if some languages had the same commands.  So a Go user could switch to the Go tab and _just stay there._  The site has a super feature where if you change tab in one place it also changes it everywhere else.  If this section, and only this section, had a "Rust/Go" tab then that wouldn't sync.

File: content/spin/quickstart.md
Comment (itowlson) [reply content/spin/quickstart.md:106]: I wondered about that.  This follows the existing naming from the K/V page, so as to play nicely with the sync feature.  Both languages could make for a long tab header.  `JavaScript` might be more global - TS devs will still recognise it - but then again the examples are TS-not-JS.

My suggestion is to defer this out of this PR and do a global rename when we have a decision.

**Comment by itowlson** [comment]
> What's the considerations for not including C#/Dotnet together with the other languages?

Just that it's significantly less ready.

File: content/spin/quickstart.md
Comment (itowlson) [reply content/spin/quickstart.md:93]: I updated it to not show other language templates but to have a note explaining users would see them.

image

(and similarly on the Go tab)

**Review by karthik2804 — APPROVED** [review]

**Comment by itowlson** [comment]
Given the approvals I am going to merge this with a couple of fixups ~so I don't have to deal with `data.json` again~.  If folks want further changes we can definitely keep iterating on it!

**Pull request merged**
